Mathlib/GroupTheory/SpecificGroups/Dihedral.lean

```diff
@@ -147,24 +151,19 @@ theorem r_one_pow (k : ℕ) : (r 1 : DihedralGroup n) ^ k = r k := by
 theorem r_one_zpow (k : ℤ) : (r 1 : DihedralGroup n) ^ k = r k := by
   cases k <;> simp
 
--- @[simp] -- Porting note: simp changes the goal to `r 0 = 1`. `r_one_pow_n` is no longer useful.
 theorem r_one_pow_n : r (1 : ZMod n) ^ n = 1 := by
-  rw [r_one_pow, one_def]
-  congr 1
-  exact ZMod.natCast_self _
+  simp
 
--- @[simp] -- Porting note: simp changes the goal to `r 0 = 1`. `sr_mul_self` is no longer useful.
-theorem sr_mul_self (i : ZMod n) : sr i * sr i = 1 := by rw [sr_mul_sr, sub_self, one_def]
+theorem sr_mul_self (i : ZMod n) : sr i * sr i = 1 := by
+  simp
 
 /-- If `0 < n`, then `sr i` has order 2.
 -/
 @[simp]
 theorem orderOf_sr (i : ZMod n) : orderOf (sr i) = 2 := by
   apply orderOf_eq_prime
   · rw [sq, sr_mul_self]
-  · -- Porting note: Previous proof was `decide`
-    revert n
-    simp_rw [one_def, ne_eq, reduceCtorEq, forall_const, not_false_eq_true]
+  · simp [← r_zero]
```

Comment:
Does `decide` work here? If not, can we isolate what the regression in `decide` is?

Reply:
To me this doesn't look like a goal that `decide` should be solving, especially after the revert.
My guess is that `decide` was applying `no_confusion` under the hood. I'm not sure that is its job.
So I propose wontfix.